labelbox/data/serialization/ndjson/converter.py

```diff
@@ -40,8 +42,34 @@ def serialize(
         Returns:
             A generator for accessing the ndjson representation of the data
         """
+
+        for label in labels:
+            annotation_uuid_lookup = {}
+            # UUIDs are private properties used to enhance UX when defining relationships.
+            # They are created for all annotations, but only utilized for relationships.
+            # To avoid overwriting, UUIDs must be unique across labels.
+            # Non-relationship annotation UUIDs are dropped (server-side generation will occur).
+            # For relationship annotations, new UUIDs are generated and stored in a lookup table.
+            for annotation in label.annotations:
+                if isinstance(annotation, RelationshipAnnotation):
+                    source_uuid = annotation.value.source._uuid
+                    target_uuid = annotation.value.target._uuid
+                    new_source_uuid = str(uuid.uuid4())
+                    new_target_uuid = str(uuid.uuid4())
+                    annotation_uuid_lookup[source_uuid] = new_source_uuid
+                    annotation_uuid_lookup[target_uuid] = new_target_uuid
+                    annotation.value.source._uuid = new_source_uuid
+                    annotation.value.target._uuid = new_target_uuid
+
+            for annotation in label.annotations:
```

Comment:
A small optimization: Can you filter out these annotations in the first pass into a separate array, then just iterate over that array? Something like:

```
annotations_to_fix = []
for annotation in label.annotations:
  if isinstance:
      ....this stuff...
  else:
    annotations_to_fix.append(annotation)

for annotation in annotation_to_fix:
  ....
```
It avoids repeating the condition logic and avoids iterating over the entire array again for a tiny cost in memory.